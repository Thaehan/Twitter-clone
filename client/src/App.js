import React, { useState } from 'react';
import { View, StyleSheet, StatusBar, TextInput, Button } from 'react-native';
import { Provider } from 'react-redux';

import RootNavigator from './navigation/RootNavigator';
import store from './redux/store';
import { createUser, getUser } from './api/user';

export default function App() {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const buttonHandle = () => {
    createUser(username, password)
      .then((res) => console.log(res.data))
      .catch((error) => console.log(error));
  };

  return (
        // <Provider store={store}>
    //   <RootNavigator />
    // </Provider>

    <View style={styles.container}>
      <StatusBar style="auto" />
      <TextInput
        style={styles.textInput}
        value={username}
        onChangeText={setUsername}
      />
      <TextInput
        value={password}
        onChangeText={setPassword}
        style={styles.textInput}
      />
      <Button title="Create user" onPress={buttonHandle} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textInput: {
    height: 50,
    backgroundColor: 'red',
  },
});

