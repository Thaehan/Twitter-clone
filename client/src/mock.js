import { logo } from './constants/ImageAssets';

const tweets = {
  avatar: logo,
  fullname: 'DoDat',
  username: 'Thaehan',
  text: 'Troi oi cai quan que gi zay ba',
};


export { tweets };
