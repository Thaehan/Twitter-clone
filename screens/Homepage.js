import { View } from 'react-native';

import HelloWorld from '../components/HelloWorld';
import Counting from '../components/Counting';
import SearchFilter from '../components/SearchFilter';

export default function Homepage() {
  return (
    <View>
      <HelloWorld/>
      <Counting/>
      <SearchFilter/>
    </View>
  )
}