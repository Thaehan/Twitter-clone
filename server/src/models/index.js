const dbConfig = require('../config/config.js');

const mongoose = require('mongoose');
mongoose.Promise = global.Promise;
mongoose.set('useNewUrlParser', true);
mongoose.set('useFindAndModify', false);
mongoose.set('useCreateIndex', true);
const db = {};
db.mongoose = mongoose;
db.url = dbConfig.url;

db.tutorials = require('./tutorial.model')(mongoose);
db.users = require('./user.model')(mongoose);

module.exports = db;
