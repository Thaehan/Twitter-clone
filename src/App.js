import { View, Text } from 'react-native';
import React from 'react';
import RootNavigator from './navigation/RootNavigator';

export default function App() {
  return <RootNavigator />;
}
