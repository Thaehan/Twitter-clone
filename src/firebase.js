import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import { getFirestore } from 'firebase/firestore/lite';
import { getStorage } from 'firebase/storage/';

const config = {
  apiKey: 'AIzaSyDZD0dNa5P0RrkliqTVSQALpBOfNfFZOtc',
  authDomain: 'twitter-clone-5bfb8.firebaseapp.com',
  projectId: 'twitter-clone-5bfb8',
  storageBucket: 'twitter-clone-5bfb8.appspot.com',
  messagingSenderId: '411924634090',
  appId: '1:411924634090:web:90d88045e92680d2eaba93',
  measurementId: 'G-C6C5W57VVZ',
};

const app = initializeApp(config);
const auth = getAuth(app);
const db = getFirestore(app);
const storage = getStorage(app);

export { app, auth, db, storage };
