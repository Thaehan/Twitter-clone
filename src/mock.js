import { logo, avatar } from './constants/ImageAssets';
import avatar0 from './assets/avatar0.png'
import avatar1 from './assets/avatar1.png'
import avatar2 from './assets/avatar2.png'
import avatar3 from './assets/avatar3.png'
import avatar4 from './assets/avatar4.png'
import banner0 from './assets/banner0.png'
import banner1 from './assets/banner1.png'
import banner2 from './assets/banner2.png'
import banner3 from './assets/banner3.png'

const tweets = {
  "tweetId": "850006245121695744",
  "userPosted": "1",
  "textContent": "1/ Today we\u2019re sharing our vision for the future of the Twitter API platform! @Nowano #Loach!\nhttps://t.co/XweGngmxlP",
  "mediaContent": "assets/black",
  "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
  "referedPostId": "",
  "userMentioned": ["5"]
};
const tweetsList = [
  {
    "tweetId": "0",
    "userPosted": "0",
    "textContent": "Good morning!! How everyone doing today?",
    "mediaContent": "assets/black",
    "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
    "referedPostId": "",
    "userMentioned": ["2"]
  },
  {
    "tweetId": "1",
    "userPosted": "1",
    "textContent": "Layout direction specifies the direction in which children and text in a hierarchy should be laid out. Layout direction also affects what edge start and end refer to. By default, React Native lays out with LTR layout direction. In this mode start refers to left and end refers to",
    "mediaContent": "assets/black",
    "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
    "referedPostId": "",
    "userMentioned": []
  },
  {
    "tweetId": "2",
    "userPosted": "2",
    "textContent": "Any body show me a good nft place?",
    "mediaContent": "assets/black",
    "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
    "referedPostId": "",
    "userMentioned": []
  },
  {
    "tweetId": "3",
    "userPosted": "3",
    "textContent": ":D",
    "mediaContent": "assets/black",
    "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
    "referedPostId": "",
    "userMentioned": []
  }
]
const userDatabase = [
  {
    userId: "0",
    username: "Đỗ Đạc",
    handle: "thaehan1998",
    avatar: avatar0,
    banner: banner0,
    bio: "I am Đạt",
    following: [],
    liked: [],
  },
  {
    userId: "1",
    username: "Danh",
    handle: "wowyboi",
    avatar: avatar1,
    banner: banner1,
    bio: "I am Danh",
    following: [],
    liked: [],
  },
  {
    userId: "2",
    username: "Snoop Cat",
    handle: "snoppKitty",
    avatar: avatar2,
    banner: banner2,
    bio: "Smoke catweed everyday",
    following: [],
    liked: [],
  },
  {
    userId: "3",
    username: "Uyennhu",
    handle: "uwu",
    avatar: avatar3,
    banner: banner3,
    bio: "Sike fam",
    following: [],
    liked: [],
  },
]



const thaehan = {
  username: "Nick Ao Do Ven Dat",
  handle: "thaehan1998",
  password: "5",
  avatar: avatar,
  userId: "1"
}

export { tweets, thaehan, tweetsList, userDatabase };
