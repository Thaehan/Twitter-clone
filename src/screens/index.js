import Login from './Login';
import Signup from './Signup';
import Home from './Home';

export { Login, Signup, Home };
