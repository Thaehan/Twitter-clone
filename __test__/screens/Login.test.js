import React from 'react';
// import renderer from 'react-test-renderer';\
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import { Login } from '../../src/screens';
import store from '../../src/redux/store';

jest.useFakeTimers();
jest.mock(
  'react-native/Libraries/Animated/NativeAnimatedHelper'
);

describe('<Login />', () => {
  it('Render Login', () => {
    const loginScreen = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    expect(loginScreen.toJSON()).toMatchSnapshot();
  });
});
