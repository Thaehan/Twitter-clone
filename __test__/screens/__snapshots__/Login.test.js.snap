// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Render Login 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "fontSize": 16,
      },
      Object {
        "flex": 2,
        "width": 414,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 88,
        "marginBottom": 66,
        "marginLeft": 163,
        "width": 88,
      }
    }
  />
  <Text
    style={
      Object {
        "fontWeight": "500",
        "marginBottom": 30,
        "marginLeft": 30,
      }
    }
  >
    Login to Twitter
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="#BDBDBD"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#f6f6f6",
        "borderColor": "#f8f8f8",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 50,
        "marginBottom": 20,
        "marginLeft": 30,
        "paddingLeft": 16,
        "width": 350,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    placeholderTextColor="#BDBDBD"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#f6f6f6",
        "borderColor": "#f8f8f8",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 50,
        "marginBottom": 20,
        "marginLeft": 30,
        "paddingLeft": 16,
        "width": 350,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "borderRadius": 50,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontWeight": "500",
                "padding": 0,
              },
              Object {
                "color": "#1da1f2",
              },
            ]
          }
        >
          Forgot your password?
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": 271,
        "marginTop": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#1da1f2",
          "borderRadius": 100,
          "fontWeight": "500",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "paddingBottom": 8,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 6,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 85,
        "marginTop": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "400",
          "marginLeft": 15,
          "marginRight": 8,
        }
      }
    >
      Do not have an account?
    </Text>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "borderRadius": 50,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontWeight": "500",
                "padding": 0,
              },
              Object {
                "color": "#1da1f2",
              },
            ]
          }
        >
          Sign up
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
